### PR TITLE
docs: added taxes conceptual guide

### PR DESCRIPTION
**Pull request by shahednasser**
Added a taxes conceptual guide.

Closes DOCS-508

### CODE REVIEW

File: docs/content/modules/taxes/taxes.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The `TaxRate` entity’s attributes include:
+
+- `rate`: a number indicating the tax rate to apply for the specified conditions.
+- `code`: a string indicating the tax code of this rate.
+- `name`: a string indicating the name of the tax rate.
+- `region_id`: a string indicating the ID of the region this tax rate applies to. You can also access the region by expanding the `region` relation and accessing `tax_rate.region`.
+
+### Tax Provider
+
+Tax providers are used to return the tax line items for a set of line items and shipping methods during checkout. The Medusa backend has a `system` tax provider. It returns the line items as-is, without making any changes. You can also create your own tax provider that matches your use case.
```
Comment (olivermrbl) [inline docs/content/modules/taxes/taxes.md:55]: suggestion: Add an example here. Not as code but to explain the concept. An easy yet very relevant one would be fetching tax lines based on an address.

**Review by olivermrbl — APPROVED** [review]
Very nice! Added a minor suggestion, but otherwise LGTM 👍

**Review by olivermrbl — APPROVED** [review]
Very nice! Added a minor suggestion, but otherwise LGTM 👍

File: docs/content/modules/taxes/taxes.md (hunk 1 above)
Comment (shahednasser) [reply docs/content/modules/taxes/taxes.md:55]: Would something like this work?

```ts
const taxLines = await this.taxProviderService_
          .withTransaction(this.activeManager_)
          .getTaxLines(lineItems,{
            shipping_address: {
              city,
              // etc...
            }
          })

Comment (olivermrbl) [reply docs/content/modules/taxes/taxes.md:55]: I would actually add it without code and just as a very minor comment:

"As an example, you might want to use the shipping address of the cart to construct the tax lines."

**Pull request merged**